src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -9,11 +9,144 @@
 
 package common
 
-import "fmt"
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/cmd/container/common"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"golang.org/x/sync/errgroup"
+)
 
 const FeedbackMsg = `We'd love to hear about your experience with redpanda:
 https://vectorized.io/feedback`
 
 func DeprecationMessage(altCmd string) string {
 	return fmt.Sprintf("use '%s' instead.", altCmd)
 }
+
+// Try to read the config from the default expected locations, or from the
+// specific path passed with --config. If --config wasn't passed, and the config
+// wasn't found, return the default configuration.
+func FindConfigFile(
+	mgr config.Manager, configFile *string,
+) func() (*config.Config, error) {
+	var conf *config.Config
+	var err error
+	return func() (*config.Config, error) {
```

Comment:
I'm not a golang dude, but why return func()?

Reply:
This is the best way I could explain it when I did this: https://github.com/vectorizedio/redpanda/blob/dev/src/go/rpk/pkg/cli/cmd/api.go#L53-L68